### PR TITLE
Add a local-only "play-test" mode

### PR DESCRIPTION
**Pull request by rhelmer**
Depends on https://github.com/mozilla-rally/rally-sdk/pull/15

This adds a super basic "play-test" mode, which is activated when the developer mode flag is passed (`npm run dev` does this), and adds a built-in "options" HTML page to the study-template extension:

image

This causes the extension to start up and never attempt to contact Firebase or the Rally website, instead it listens for messages from the buttons above. The idea is that you can control the lifecycle of the study, and export data, without having to use Rally Web Platform or set up an emulator.

### CODE REVIEW

**Comment by rhelmer** [comment]
This implements a very basic UI as an extension "options page" (see https://github.com/mozilla-rally/study-template/pull/175#issue-998896427 for screenshot), a toolbar button to get to the UI, and a download button to export data (this just dumps the contents of the browser extension local storage area, and prompts the user to download it as a JSON file).

Tailwind is used for CSS, for the moment it's just bundling the full default tailwind CSS since the UI is plain HTML/JS and isn't being put through a bundler yet. Once we move the UI to a fancier framework, we can have `rollup` tree-shake out unused styles.

We're also going to want to improve the data export feature here (we'll want to integrate better with Glean and provide the ability to download datasets in a way that'll easily load into tools like R and Jupyter, even if it's just CSVs).

I think this is a good enough starting point though. I'll file follow-up issues for the above.

**Comment by rhelmer** [comment]
I'm going to package rally-sdk properly and publish on npm (we need https://github.com/mozilla-rally/rally-sdk/pull/15 for this PR) and will re-request review once that's done.

File: README.md [hunk 1]
```diff
@@ -48,20 +48,18 @@ Prerequisites: current versions of [Firefox](https://www.mozilla.org/firefox/new
...
+* `dev:emulator` - Like `dev`, but run in _emulator mode_, which connects to a local Firebase emulator. See the [Rally Web Platform docs](https://github.com/mozilla-rally/rally-web-platform#quickstart) for information on running a local Firebase emulator.
 * `lint`: Run linting on the study extension.
 * `package`: Build the study extension (`build`), then package the built study into an archive for distribution. Output is in the `web-ext-artifacts/` directory.
-* `start`: Build the study extension (`build`), then launch Firefox with the study extension installed.
-* `start-dev`: Build the study extension in _developer mode_ (`dev`), then launch Firefox with the study extension installed.
 * `test-integration`: Packages the study extension (`package`), then runs the provided integration test.
```
Comment (Carla-Moz) [inline README.md:62]: test:integration ?

**Review by Carla-Moz — COMMENTED** [review]
I added some suggestions to improve the UI, nothing major. Please feel free to apply or not. Also, the extension works on my end too.   

Screen Shot 2021-09-29 at 3 03 51 PM

**Comment by rhelmer** [comment]
OK thanks! Currently looks like this:

image

I do plan to move this to a framework and use standard components for it, so I don't want to iterate *too* much on it yet. I'm anticipating that studies will heavily modify / override it.

**Review by Carla-Moz — APPROVED** [review]
Agreed!

**Pull request merged**